dvc/output/base.py

```diff
@@ -362,6 +373,10 @@ def checkout(
                 filter_info=filter_info,
                 **kwargs,
             )
+            if self.isexec:
+                st = os.stat(self.path_info.fspath)
```

Comment:
We are using >=3.6, where all os methods are totally happy with Path-like objects.
```suggestion
                st = os.stat(self.path_info)
```